### PR TITLE
Add latex support with marked-katex-extension

### PR DESCRIPTION
**Pull request by julien-blanchon**
Add the latex support (https://github.com/huggingface/chat-ui/issues/446) by leveraging the [marked-katex-extension](https://github.com/UziTech/marked-katex-extension).

Screenshot 2023-09-18 at 23 32 12
Screenshot 2023-09-18 at 23 32 48

Screenshot 2023-09-18 at 23 32 27

This don't work in codeblock (for every format): This is expected and good

Screenshot 2023-09-18 at 23 35 08

I didn't try every latex format.

Math equation work well with ChatGPT they may have:
- Include a preprompt to get a specific format and tell the LLM to format as latex
- Make some postprocess (not likely)

### CODE REVIEW

File: src/lib/components/chat/ChatMessage.svelte
Comment (gary149) [inline src/lib/components/chat/ChatMessage.svelte:70]: Do you know what breaks? We'll probably want to keep the same options as before.https://github.com/markedjs/marked/blob/2ec4c00966844abc86c4d5f6106d71ba7b8b9f82/src/defaults.ts

Comment (julien-blanchon) [reply src/lib/components/chat/ChatMessage.svelte:70]: I will investigate, looking at your link this is likely `extensions: null,`. We may add the markedKatex extension directly in the config or just get rid of the .getDefaults and manually import the same config

Comment (julien-blanchon) [reply src/lib/components/chat/ChatMessage.svelte:70]: Yes extensions is breaking markedKatex. One workaround is this:

```typescript
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const { extensions, ...defaults } = marked.getDefaults() as marked.MarkedOptions & {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    extensions: any;
};
const options: marked.MarkedOptions = {
    ...defaults,
    gfm: true,
    breaks: true,
    renderer,
};
```

I did add some typing and eslint-disable

**Comment by julien-blanchon** [comment]
I think this one is ready to merge, I'm waiting for review

**Review by gary149 — APPROVED** [review]
Ty! LGTM, I'll let @nsarrazin take a look then merge 👍

**Review by julien-c — COMMENTED** [review]
Wow, this PR looks really cool!

**Review by nsarrazin — APPROVED** [review]
I tested it locally in detail, looks really cool, thanks for doing this! :fire: 

The [dep itself](https://github.com/UziTech/marked-katex-extension) doesn't seem super popular (20 stars, 300weekly downloads) but I checked the code and everything looks clean there too so I think it's fine.

Thanks for the great contribution, merging this now!

**Comment by nsarrazin** [comment]
Your changes are live in prod at https://huggingface.co/chat :smile: Thanks! @julien-blanchon

**Comment by julien-blanchon** [comment]
> Your changes are live in prod at https://huggingface.co/chat 😄 Thanks! @julien-blanchon

Thank you very much! I'm very happy to have contributed

**Pull request merged**
